### ISSUE TITLE
Enabling Push Notifications for Travis-CI mobile apps

### ISSUE THREAD

**Issue by floydpink** [travis-ci/travis-ci#1258]
**tl;dr: Could every Travis-CI build make a POST call with the repository id and build status to enable mobile push notifications?**

The [Travis-CI Mobile android app](https://play.google.com/store/apps/details?id=com.floydpink.android.travisci) has been published at Google Play store for a while now, with a port of [a limited subset of the features](http://floydpink.github.io/Travis-CI/features.html) from [the official web client](https://travis-ci.org/). 

The [android app](https://github.com/floydpink/Travis-CI) itself is done in PhoneGap with the embedded web app maintained as [a separate repository (git submodule)](https://github.com/floydpink/Travis-CI-www) so as to enable reusing this for other mobile platforms that has PhoneGap support.

I have started off with [an iOS flavor](https://github.com/floydpink/Travis-CI-iOS) and also published the current Android version to the iOS App Store, to be rejected by Apple citing lack of iOS features as the prime reason.

I am trying to enhance the look & feel a bit and also to try and include Push notifications for both Android and iOS clients. 

For push notifications I have written a lightweight [server component](https://github.com/floydpink/travis-notification-hub) in node.js/MongoDB, that would register and keep track of the devices and the repositories the devices subscribes to. The app will work in a non-authenticated mode, where any repositories marked as favorites are what will be subscribed for build notifications.

Following the default notification style in Travis-CI, the mobile notifications would be sent only when builds are broken and subsequently for every build until it gets fixed.

Questions:
1. General feedback - how does this all sound?
2. Travis-CI integration - if it sounds good, could we make the changes to do an http POST with minimal payload like `repoId` and build status on every build's completion?
3. Hosting guidance - I am also looking for help/guidance in deciding where to host the server component - have looked at the free plans on nodejitsu and heroku till now and they seem workable to me, but since I do not have much experience there, would like someone comment on the size and scale requirements.

Thank you very much for reading this. :smile:
